### PR TITLE
Prettier pre commit hook

### PR DESCRIPTION
**Pull request by AGS-**

### CODE REVIEW

**Comment by AGS-** [comment]
Please squash

File: package.json [hunk 1]
```diff
@@ -74,9 +81,12 @@
     "express-graphql": "0.6.6",
     "flow-bin": "0.47.0",
     "graphql": "^0.10.1",
+    "husky": "^0.13.4",
```
Comment (wincent) [inline package.json]: #lazyweb: How does husky actually work? Is it literally installing a precommit hook in the local repo?

Comment (AGS-) [inline package.json]: Yeah, pretty much.

Comment (asiandrummer) [inline package.json]: Feel like this should be a `devDependencies` entry

File: package.json [hunk 2]
```diff
@@ -36,11 +36,18 @@
...
+    "precommit": "lint-staged",
     "prepublish": ". ./resources/prepublish.sh",
     "preversion": ". ./resources/checkgit.sh && npm test",
     "test": "npm run lint && npm run check && npm run testonly && npm run build",
     "testonly": "babel-node ./node_modules/.bin/_mocha $npm_package_options_mocha"
   },
+  "lint-staged": {
+    "*.js": [
+      "prettier --jsx-bracket-same-line --single-quote --trailing-comma=all --write",
+      "git add"
```
Comment (asiandrummer) [inline package.json]: let's also run `npm test` to make sure we didn't break anything, although `prettier` shouldn't have in the first place.

**Comment by asiandrummer** [comment]
I'm approving this to make merging this PR easier

File: package.json (hunk 2 above)
Comment (AGS-) [inline package.json]: oh yeah, I think it would be pretty good to run `test` just in case

File: package.json (hunk 1 above)
Comment (AGS-) [inline package.json]: It is, it just got folded right in between dependencies and dev dependencies in the diff view 😛

**Comment by astorije** [comment]
Hi there! Should this actually update the commit, or warn the committer? I just met an issue when opening #568 that I understand may have to do with this, [explained here](https://github.com/graphql/graphiql/pull/568#issuecomment-321988358).
Commenting here to keep track of the related issue. Thanks for your help and work :)

**Comment by wincent** [comment]
@astorije: I'll reply on #568.

**Pull request merged**